crates/core/tedge_mapper/src/az/mapper.rs

```diff
@@ -21,6 +22,28 @@ impl AzureMapper {
 
 #[async_trait]
 impl TEdgeComponent for AzureMapper {
+    fn session_name(&self) -> &str {
+        AZURE_MAPPER_NAME
+    }
+
+    async fn init(&self) -> Result<(), anyhow::Error> {
+        info!("Initialize tedge mapper az");
+        create_directory_with_user_group(
+            "/etc/tedge/operations/az",
+            "tedge-mapper",
+            "tedge-mapper",
+            0o775,
+        )?;
+        mqtt_channel::init_session(&self.get_mqtt_config()?).await?;
```

Comment:
This is buggy. Since the configuration returned by `self.get_mqtt_config()` doesn't set the list of subscriptions for azure.

It would be better to introduce a  `TEdgeComponent::init_session` reference implementation and to use it here.

```suggestion
    async fn init(&self) -> Result<(), anyhow::Error> {
        self.init_session(azure_topics).await?
    }
```